taxi/test/test_forms.py

```diff
@@ -0,0 +1,140 @@
+        self.assertTrue(form.is_valid())
+
+
+class TestManufacturerForms(TestCase):
+    def setUp(self):
+        self.user = get_user_model().objects.create_user(
+            username="new_driver",
+            password="root1234",
+        )
+        self.client.force_login(self.user)
+
+        self.manufacturer = Manufacturer.objects.create(
+            name="Test manufacture", country="USA"
+        )
+
+    def test_manufacturer_search_form_with_arg(self):
+        form_data = {
+            "name": "Test manufacture",
+        }
+        form = ManufacturerNameSearchForm(data=form_data)
+        self.assertTrue(form.is_valid())
+
+    def test_manufacturer_search_form_without_arg(self):
+        form_data = {
+            "name": "",
+        }
+        form = ManufacturerNameSearchForm(data=form_data)
+        self.assertTrue(form.is_valid())
+
+
+class TestDriverForms(TestCase):
+    def setUp(self):
+        self.user1 = get_user_model().objects.create_user(
+            username="new_driver",
+            license_number="HRN84739",
+            password="root1234",
+        )
+        self.user2 = get_user_model().objects.create_user(
+            username="another_driver",
+            license_number="HRN12345",
+            password="root1234",
+        )
+        self.user3 = get_user_model().objects.create_user(
+            username="some_other_driver",
+            license_number="HRN67890",
+            password="root1234",
+        )
+        self.client.force_login(self.user1)
+
+        self.manufacturer = Manufacturer.objects.create(
+            name="Test manufacture", country="USA"
+        )
+
+    def test_driver_form(self):
+        form_data = {
+            "username": "fourth_driver",
+            "license_number": "HRN84735",
+            "password1": "Root1234",
+            "password2": "Root1234"
+        }
+        form = DriverCreationForm(data=form_data)
+        self.assertTrue(form.is_valid())
+        new_driver = form.save()
+        self.assertIsNotNone(new_driver)
+        created_user = get_user_model().objects.get(username="fourth_driver")
+        self.assertEqual(created_user.username, "fourth_driver")
+        self.assertEqual(created_user.license_number, "HRN84735")
+        self.assertTrue(created_user.check_password("Root1234"))
+
+    def test_driver_search_form_with_arg(self):
+        form_data = {
+            "username": "new_driver",
+        }
+        form = DriverUsernameSearchForm(data=form_data)
+        self.assertTrue(form.is_valid())
+
+    def test_driver_search_form_without_arg(self):
+        form_data = {
+            "username": "",
+        }
+        form = DriverUsernameSearchForm(data=form_data)
+        self.assertTrue(form.is_valid())
+        search_username = form.cleaned_data.get("username")
+        results = get_user_model().objects.filter(
+            username__icontains=search_username
+        )
+        self.assertEqual(results.count(), 3)
```

Comment:
example of correct implementation of search tests would be: your search form has 'new_driver',  then you check in results if new_driver is in results and other drivers are not in result. and you need to do the same for Cars and Manufacturers